homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -367,13 +367,9 @@ def state_received(msg: ReceiveMessage) -> None:
             if brightness_supported(self.supported_color_modes):
                 try:
                     if brightness := values["brightness"]:
+                        scale = self._config[CONF_BRIGHTNESS_SCALE]
                         self._attr_brightness = min(
-                            int(
-                                brightness  # type: ignore[operator]
-                                / float(self._config[CONF_BRIGHTNESS_SCALE])
-                                * 255
-                            ),
-                            255,
+                            255, round(brightness * 255 / scale)  # type: ignore[operator]
```

Comment:
OK, so this change replaces truncating with rounding, that makes sense

Reply:
It is based on de fan percentage scaling code we use for fan. In fact the logic is not changed. Therefor the tests did not need an update.

Reply:
This way the value we send will closer result if the same value is sent and translated back.